test/performance/odp_queue_perf.c

```diff
@@ -443,13 +481,79 @@ static int run_test(void *arg)
 	stat->deq_retry = num_deq_retry;
 	stat->enq_retry = num_enq_retry;
 
-	return ret;
+	return 0;
+}
+
+static void map_queues_to_threads(test_global_t *global)
+{
+	test_options_t *opt = &global->options;
+
+	if (!opt->private_queues) {
+		for (uint32_t i = 0; i < opt->num_queue; i++) {
+			for (uint32_t j = 0; j < opt->num_cpu; j++) {
+				global->thread_args[j].src_queue_id[i] = i;
+				global->thread_args[j].dst_queue_id[i] = i;
+				global->thread_args[j].num_queues++;
+			}
+		}
+		return;
+	}
+
+	if (opt->num_queue % opt->num_cpu)
+		ODPH_ERR("Warn: %" PRIu32 " queues shared unevenly amongst %" PRIu32 " workers.\n",
+			 opt->num_queue, opt->num_cpu);
+
+	for (uint32_t i = 0; i < opt->num_queue; i++) {
+		uint32_t thread_idx = i % opt->num_cpu;
+		uint32_t queue_idx = global->thread_args[thread_idx].num_queues;
+
+		global->thread_args[thread_idx].src_queue_id[queue_idx] = i;
+		global->thread_args[thread_idx].dst_queue_id[queue_idx] = i;
+		global->thread_args[thread_idx].num_queues++;
+	}
+}
+
+static void print_queue_mappings(test_global_t *global)
+{
+	printf("Worker-queue mappings\n");
+	printf("---------------------\n");
+
+	for (uint32_t i = 0; i < global->options.num_cpu; i++) {
+		thread_args_t *thread_args = &global->thread_args[i];
+		uint32_t num_queues = thread_args->num_queues;
+
+		printf("Worker %u:\n", i);
+
+		printf("  src queue:");
```

Comment:
Maybe "src queue indexes" or something similar, as actual handle values are not printed.

Reply:
Updated in V3.